classes/user/interest/Repository.php

```diff
@@ -0,0 +1,144 @@
+     */
+    public function getInterestsForUser(User $user): array
+    {
+        return ControlledVocabEntry::query()
+            ->whereHas(
+                "controlledVocab",
+                fn($query) => $query
+                    ->withSymbolic(UserInterest::CONTROLLED_VOCAB_INTEREST)
+                    ->withAssoc(0, 0)
+            )
+            ->whereHas("userInterest", fn($query) => $query->withUserId($user->getId()))
+            ->get()
+            ->pluck(UserInterest::CONTROLLED_VOCAB_INTEREST, 'id')
+            ->toArray();
+    }
+
+    /**
+     * Returns a comma separated string of a user's interests
+     */
+    public function getInterestsString(User $user): string
+    {
+        $interests = $this->getInterestsForUser($user);
+
+        return implode(', ', $interests);
+    }
+
+    /**
+     * Set a user's interests
+     */
+    public function setInterestsForUser(User $user, string|array|null $interests = null): void
+    {
+        $interests = is_array($interests)
+            ? $interests
+            : (empty($interests) ? [] : explode(',', $interests));
+
+        $controlledVocab = Repo::controlledVocab()->build(
+            UserInterest::CONTROLLED_VOCAB_INTEREST
+        );
+
+        $currentInterests = ControlledVocabEntry::query()
+            ->whereHas(
+                'controlledVocab',
+                fn($query) => $query
+                    ->withSymbolic(UserInterest::CONTROLLED_VOCAB_INTEREST)
+                    ->withAssoc(0, 0)
+            )
+            ->withLocale('')
+            ->withSetting(UserInterest::CONTROLLED_VOCAB_INTEREST, $interests)
+            ->get();
+        
+        try {
+
+            DB::beginTransaction();
+
+            // Delete the existing interests association.
+            UserInterest::query()->withUserId($user->getId())->delete();
+            
+            $newInterestIds = collect(
+                    array_diff(
+                        $interests,
+                        $currentInterests->pluck(UserInterest::CONTROLLED_VOCAB_INTEREST)->toArray()
+                    )
+                )
+                ->map(fn (string $interest): string => trim($interest))
+                ->unique()
+                ->map(
+                    fn (string $interest) => ControlledVocabEntry::create([
+                        'controlledVocabId' => $controlledVocab->id,
+                        UserInterest::CONTROLLED_VOCAB_INTEREST => [
+                            '' => $interest
+                        ],
+                    ])->id
+                );
+            
+            collect($currentInterests->pluck('id'))
+                ->merge($newInterestIds)
+                ->each(fn ($interestId) => UserInterest::create([
+                    'userId' => $user->getId(),
+                    'controlledVocabEntryId' => $interestId,
+                ]));
+            
+            Repo::controlledVocab()->resequence($controlledVocab->id);
```

Comment:
do we need to resequence the vocabs for user interest, don't see any benefit or usage of it ?